assets/admin/students/student-modal/course-list.js

```diff
@@ -1,54 +1,109 @@
 /**
  * WordPress dependencies
  */
-import apiFetch from '@wordpress/api-fetch';
 import { CheckboxControl, Spinner } from '@wordpress/components';
-import { useEffect, useState } from '@wordpress/element';
+import { useEffect, useState, useCallback, useRef } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
+/**
+ * Internal dependencies
+ */
+import httpClient from './lib/http-client';
```

Comment:
I wonder if we should move this up into the `admin` folder?

Reply:
Let's define if we should maintain the httpClient or back to apiFetch.

Reply:
Moved [here](https://github.com/Automattic/sensei/pull/5017/commits/b04bcd170caf6f37a22e27855e0424641f6519ba)